src/Node/Expression/ParentExpression.php

```diff
@@ -28,19 +28,36 @@ public function __construct(string $name, int $lineno, string $tag = null)
 
     public function compile(Compiler $compiler): void
     {
-        if ($this->getAttribute('output')) {
-            $compiler
-                ->addDebugInfo($this)
-                ->write('$this->displayParentBlock(')
-                ->string($this->getAttribute('name'))
-                ->raw(", \$context, \$blocks);\n")
-            ;
+        if ($compiler->getEnvironment()->useYield()) {
+            if ($this->getAttribute('output')) {
+                $compiler
+                    ->addDebugInfo($this)
+                    ->write('yield from $this->yieldParentBlock(')
+                    ->string($this->getAttribute('name'))
+                    ->raw(", \$context, \$blocks);\n")
+                ;
+            } else {
+                $compiler
+                    ->raw('$this->renderParentBlock(')
```

Comment:
Will this not throw an exception (as `renderParentBlock` is forbidden inside a YieldingTemplate) ?  

cf:

```php
// src/YieldingTemplate.php
public function renderParentBlock($name, array $context, array $blocks = [])
{
    throw new RuntimeError(sprintf('Calling "%s" for block "%s" is not supported as "use_yield" is set to "true".', __METHOD__, $name), -1, $this->getSourceContext());
}
```

Reply:
Here is the case that led me there

The following test is green with `use_yield=false` and fails with `use_yield=true`

```
--TEST--
"block" calling parent() in a conditional expression
--TEMPLATE--
{% extends "parent.twig" %}
{% block label %}{{ parent() ?: 'foo' }}{% endblock %}
--TEMPLATE(parent.twig)--
{% block label %}PARENT_LABEL{% endblock %}
--DATA--
return []
--EXPECT--
PARENT_LABEL

```

Reply:
Good catch, fixed now.